src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -0,0 +1,2164 @@
+        private SafeInternetHandle _sessionHandle;
+        #endregion
+
+        public WinHttpHandler()
+        {
+        }
+
+        #region Properties
+        public AutomaticRedirectionPolicy AutomaticRedirectionPolicy
+        {
+            get
+            {
+                return _automaticRedirectionPolicy;
+            }
+
+            set
+            {
+                if (value != AutomaticRedirectionPolicy.Never
+                    && value != AutomaticRedirectionPolicy.DisallowHttpsToHttp
+                    && value != AutomaticRedirectionPolicy.Always)
+                {
+                    throw new ArgumentOutOfRangeException("value");
+                }
+
+                CheckDisposedOrStarted();
+                _automaticRedirectionPolicy = value;
+            }
+        }
+
+        public int MaxAutomaticRedirections
+        {
+            get
+            {
+                return _maxAutomaticRedirections;
+            }
+
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(
+                        "value",
+                        value,
+                        string.Format(SR.net_http_value_must_be_greater_than, 0));
+                }
+
+                CheckDisposedOrStarted();
+                _maxAutomaticRedirections = value;
+            }
+        }
+
+        public DecompressionMethods AutomaticDecompression
+        {
+            get
+            {
+                return _automaticDecompression;
+            }
+
+            set
+            {
+                CheckDisposedOrStarted();
+                _automaticDecompression = value;
+            }
+        }
+
+        public CookieUsePolicy CookieUsePolicy
+        {
+            get
+            {
+                return _cookieUsePolicy;
+            }
+
+            set
+            {
+                if (value != CookieUsePolicy.IgnoreCookies
+                    && value != CookieUsePolicy.UseInternalCookieStoreOnly
+                    && value != CookieUsePolicy.UseSpecifiedCookieContainer)
+                {
+                    throw new ArgumentOutOfRangeException("value");
+                }
+
+                CheckDisposedOrStarted();
+                _cookieUsePolicy = value;
+            }
+        }
+
+        public CookieContainer CookieContainer
+        {
+            get
+            {
+                return _cookieContainer;
+            }
+
+            set
+            {
+                CheckDisposedOrStarted();
+                _cookieContainer = value;
+            }
+        }
+
+        public SslProtocols SslProtocols
+        {
+            get
+            {
+                return _sslProtocols;
+            }
+
+            set
+            {
+                CheckDisposedOrStarted();
+
+                SslProtocols allowedSslProtocols = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
```

Comment:
No problem, I just know that some of the credit card processing guidelines are changing to disallow TLS 1.0, so wanted to make sure that it got looked at.